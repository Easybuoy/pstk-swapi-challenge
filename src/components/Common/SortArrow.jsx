import React from 'react';

const SortArrow = ({ order }) => {
  if (order === 1) {
    return <span>&uarr;</span>;
  } else if (order === 0) {
    return <span>&darr;</span>;
  } else {
    return '';
  }
};

export default SortArrow;
