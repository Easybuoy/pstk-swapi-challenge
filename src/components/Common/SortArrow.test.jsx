import React from 'react';
import { shallow } from 'enzyme';

import SortArrow from './SortArrow';

describe('<SortArrow />', () => {
  let props = {
    order: 0
  };

  it('renders the SortArrow component correctly', () => {
    const wrapper = shallow(<SortArrow {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the SortArrow component correctly', () => {
    props.order = 1;
    const wrapper = shallow(<SortArrow {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the SortArrow component correctly', () => {
    props.order = '';
    const wrapper = shallow(<SortArrow {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
