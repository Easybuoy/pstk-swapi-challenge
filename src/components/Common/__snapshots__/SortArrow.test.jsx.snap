// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SortArrow /> renders the SortArrow component correctly 1`] = `
<span>
  ↓
</span>
`;

exports[`<SortArrow /> renders the SortArrow component correctly 2`] = `
<span>
  ↑
</span>
`;

exports[`<SortArrow /> renders the SortArrow component correctly 3`] = `""`;
